release_monitor/release_monitor.py

```diff
@@ -114,6 +114,8 @@ def run(self):
         self.log.info("Registered signal handler for liveness probe")
 
         while True:
+            self.renew_rss_feeds()
```

Comment:
do I understand correctly that if the rss_feed is empty at beginning, the notification is send for all packages?

Reply:
Based on my testing. The implementation is broken at type system level. For example: the old RSS feed is not an empty list, but None, so the comparison fails in the first cycle; if you use the first feed content as an initial value for the old RSS feed, the issue is replaced by a different one. More specifically the "entry" type does not implement comparison, so you cannot sort it. If you remove the sorting, it works again, but there is a chance that the feed itself is not sorted, so the comparison might give undefined result.

Long story short, it is hard to reason about this code :-).